ocp_resources/machineset.py

```diff
@@ -0,0 +1,195 @@
+        machine_role (str): machine role. e.g.: 'worker'.
+        machine_type (str): machine role. e.g.: 'worker'.
+        replicas (int): amount of replicas the machine-set will have.
+        provider_spec (dict): provider spec information.
+
+        provider spec example:
+
+    {
+        "value": {
+            "apiVersion": "ovirtproviderconfig.machine.openshift.io/v1beta1",
+            "auto_pinning_policy": "none",
+            "cluster_id": 5612af70-f4f5-455d-b7df-fbad66accc38,
+            "cpu": {
+                "cores": 8,
+                "sockets": 1,
+                "threads": 1
+            },
+            "credentialsSecret": {
+                "name": "ovirt-credentials"
+            },
+            "kind": "OvirtMachineProviderSpec",
+            "memory_mb": 16000,
+            "os_disk": {
+                "size_gb": 31
+            },
+            "template_name": "ge2n1-gcwmg-rhcos",
+            "type": "server",
+            "userDataSecret": {
+                "name": "worker-user-data"
+            }
+        }
+    }
+    """
+
+    api_group = NamespacedResource.ApiGroup.MACHINE_OPENSHIFT_IO
+
+    def __init__(
+        self,
+        cluster_name=None,
+        name=None,
+        namespace=None,
+        teardown=True,
+        client=None,
+        machine_role="worker",
+        machine_type="worker",
+        replicas=1,
+        provider_spec=None,
+        yaml_file=None,
+    ):
+        super().__init__(
+            name=name,
+            namespace=namespace,
+            client=client,
+            teardown=teardown,
+            yaml_file=yaml_file,
+        )
+        self.replicas = replicas
+        self.cluster_name = cluster_name
+        self.machine_role = machine_role
+        self.machine_type = machine_type
+        self.provider_spec = provider_spec or {}
+
+    @property
+    def available_replicas(self):
+        return self.instance.status.availableReplicas
+
+    @property
+    def ready_replicas(self):
+        return self.instance.status.readyReplicas
+
+    @property
+    def desired_replicas(self):
+        return self.instance.status.replicas
+
+    @property
+    def provider_spec_value(self):
+        return self.instance.spec.template.spec.providerSpec.value
+
+    def wait(self, timeout=TIMEOUT, sleep=1):
+        """
+        Wait for machine-set to reach 'ready' state.
+
+        Args:
+            timeout (int): maximum time to wait for the 'ready' state.
+            sleep (int): sleep time between each sample.
+
+        Returns:
+            bool: True if machine-set reached 'ready' state, False otherwise.
+        """
+        try:
+            for ready_replicas in TimeoutSampler(
+                wait_timeout=timeout, sleep=sleep, func=lambda: self.ready_replicas
+            ):
+                if ready_replicas and ready_replicas == self.desired_replicas:
+                    return True
+        except TimeoutExpiredError:
+            LOGGER.error(
```

Comment:
please fit all errors in one LOGGER.error

Reply:
ack